### PR TITLE
disallowing access to git folders

### PR DESCRIPTION
**Pull request by sunnysideup**
Just in case anyone ever forgets.

### CODE REVIEW

**Comment by dhensby** [comment]
Really all `.` folders should be denied access to (except `.well-known`) out of the box. My understanding was that this was a pretty standard setting for webfolders.

That, plus you really shouldn't have a `.git` in production at all means that this rule should be redundant and (if it's not) then there's a huge list of other potential files we could need to block.

For things like this I go to look at h5bp and see what they're doing which [is this](https://github.com/h5bp/html5-boilerplate/blob/5.3.0/dist/.htaccess#L507-L528)

To be honest I'm a little surprised we aren't doing this and maybe we should pull this in?

**Comment by assertchris** [comment]
Ideally, having a `public` folder as web root is 💯

**Comment by sunnysideup** [comment]
@dhensby I agree that all hidden files and folders should be inaccessible. I can tell you from experience that by default many webservers do allow access to these files ;-)

While I agree that .git folders should not be on production, you will find that many non-enterprise level developers will have .git folders on their live systems. 

@axllent - what are your thoughts on this (also crediting you for raising this issue).

**Comment by axllent** [comment]
@sunnysideup I fully agree, and we do block this on all our sites by default as we see regular attempts to directly access `.git/config` via the URL. Regardless whether git should or shouldn't be used for production (and why not?), I think it should be blocked by default in the default .htaccess file as it is the most common VCS these days.

**Comment by dhensby** [comment]
OK - well if this PR is update to be inline with @h5pb and on branch 3.2 then I'll accept it

**Comment by dhensby** [comment]
please rebase :)

**Comment by tractorcow** [comment]
> My understanding was that this was a pretty standard setting for webfolders.

This only SHOULD be, but isn't.

File: .htaccess [hunk 1]
```diff
@@ -40,6 +40,8 @@ ErrorDocument 500 /assets/error-500.html
 	SetEnv HTTP_MOD_REWRITE On
 	RewriteEngine On
 
+        RewriteRule \.git - [F]
```
Comment (tractorcow) [inline .htaccess]: How about `regexp for any root or sub-folder that begins with a period`. Would it be `(^|/)\.` ?

Comment (dhensby) [inline .htaccess]: I pasted this as my acceptable fix: https://github.com/h5bp/html5-boilerplate/blob/5.3.0/dist/.htaccess#L507-L528

Comment (tractorcow) [inline .htaccess]: Close!

https://github.com/h5bp/html5-boilerplate/blob/5.3.0/dist/.htaccess#L527

Comment (dhensby) [inline .htaccess]: But where is the cigar?

Comment (SpiritLevel) [inline .htaccess]: Will this do ? 

:smoking:

**Comment by sunnysideup** [comment]
I will rebase, etc... can I ask why this is done on 3.2 and not on Master?

**Comment by dhensby** [comment]
> I will rebase, etc... can I ask why this is done on 3.2 and not on Master?

master is the active development of 4.0 - so this fix wouldn't get into the wild for quite sometime and I'd prefer it to be out on the 3.x versions if possible

**Pull request closed without merging**